rust/operator-binary/src/hdfs_controller.rs

```diff
@@ -69,105 +72,100 @@ const DOCKER_IMAGE_BASE_NAME: &str = "hadoop";
 #[derive(Snafu, Debug, EnumDiscriminants)]
 #[strum_discriminants(derive(IntoStaticStr))]
 pub enum Error {
```

Comment:
note: We might want to reduce the number of variants in this error by splitting functionality up into modules. But this is out of scope for this PR.

Reply:
Yeah I agree, the larger erorr enums will tend to have groups of variants which should naturally lead to modules of errors.

Reply:
We can also put the Result type alias into said modules, so it is still easy:

```rust
fn foo() -> mod_a::Result<Bloo> { ...}
fn bar() -> mod_b::Result<Blah> { ...}
```